src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -528,5 +528,53 @@ private static Vector<byte> GetVector(byte vectorByte)
                                                        0x01ul << 48) + 1;
         private const ulong byteBroadcastToUlong = ~0UL / Byte.MaxValue;
         private const ulong filterByteHighBitsInUlong = (byteBroadcastToUlong >> 1) | (byteBroadcastToUlong << (sizeof(ulong) * 8 - 1));
+
+
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsSliceOf<T>(this Span<T> childSpan, Span<T> parentSpan) where T : struct
+        {
+            int ignored;
+            return childSpan.Length <= parentSpan.Length // avoid pointer work if trivially false
+                && IsSliceOf(ref childSpan.DangerousGetPinnableReference(), childSpan.Length,
+                ref parentSpan.DangerousGetPinnableReference(), parentSpan.Length, out ignored);
+        }
+
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsSliceOf<T>(this Span<T> childSpan, Span<T> parentSpan, out int start) where T : struct
+        {
+            start = -1;
+            return childSpan.Length <= parentSpan.Length // avoid pointer work if trivially false
+                && IsSliceOf(ref childSpan.DangerousGetPinnableReference(), childSpan.Length,
+                ref parentSpan.DangerousGetPinnableReference(), parentSpan.Length, out start);
+        }
+
+        private static unsafe bool IsSliceOf<T>(ref T child, int childCount, ref T parent, int parentCount, out int start) where T : struct
+        {
+            // compare the byte addresses for validity
+            var childStart = Unsafe.AsPointer(ref child);
+            var childEnd = Unsafe.AsPointer(ref Unsafe.Add(ref child, childCount));
+            var parentStart = Unsafe.AsPointer(ref parent);
```

Comment:
My error was to move some code that used `DangerousGetPinnableReference` into the calling method rather than using a `ref` parameter, which had very different semantics. That method is well named!